### PR TITLE
Explicit check for 201 return code when triggering downstream job.

### PR DESCRIPTION
**Pull request by xvzcf**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -134,11 +134,14 @@ jobs:
...
-                 -X POST \
+            curl --silent \
+                 --write-out "\n%{response_code}\n" \
+                 --user ${BUILD_TRIGGER_TOKEN}: \
+                 --request POST \
                  --header "Content-Type: application/json" \
                  -d '{ "branch": "main", "parameters": { "new_openssl_commit": true } }' \
-                 https://circleci.com/api/v2/project/gh/open-quantum-safe/oqs-demos/pipeline
+                 https://circleci.com/api/v2/project/gh/open-quantum-safe/oqs-demos/pipeline | tee curl_out \
+            && grep -q "201" curl_out
```
Comment (baentsch) [inline .circleci/config.yml:144]: If we now check for return code 201 as per https://circleci.com/docs/api/v2/#operation/triggerPipeline wouldn't it be prudent to also check for the returned state (as not `errored`)?

**Review by baentsch — APPROVED** [review]
Surely much better check than before; see comment as to whether it might be worth while also checking the response JSON for a proper start code: Without a build token I cannot judge/propose myself which codes are actually reasonable, so only can go by the CCI documentation.

File: .circleci/config.yml (hunk 1 above)
Comment (xvzcf) [reply .circleci/config.yml:144]: Looking at some of the responses [here](https://app.circleci.com/pipelines/github/open-quantum-safe/liboqs/1573/workflows/d7ca1a83-a26f-455c-a249-58d5ec9f102f/jobs/14633) or [here](https://app.circleci.com/pipelines/github/open-quantum-safe/openssl/825/workflows/26d72f4f-7678-4195-bd14-fb3da4efe1d6/jobs/4154) for example, the usual returned state seems to be "pending" (the transition flow is described [here](https://discuss.circleci.com/t/pipeline-states/39687/2)).

I'm not completely certain, but my guess is that if the created pipeline does error out, that should cause a job failure and subsequent e-mail notification by CircleCI.

Comment (baentsch) [reply .circleci/config.yml:144]: True. Resolved then.

**Pull request merged**